docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,273 @@
+
+- `Device.systemName: string`
+
+  Gets the device OS name.
+
+- `Device.deviceId: string`
+
+  Gets the device ID.
+
+- `Device.totalMemory: number`
+
+  Gets the device total memory, in bytes.
+
+- `Device.uniqueId: string`
+
+  Gets the device unique ID.
+
+- `Device.isTablet: boolean`
+
+  Tells if the device is a tablet.
+
+- `Device.deviceType: string`
+
+  Returns the device's type as a string, which will be one of:
+
+  - `Handset`
+  - `Tablet`
+  - `Tv`
+  - `Unknown`
+
+- `Device.supportedABIs: string[]`
+
+  Returns a list of supported processor architecture version
+
+  **Examples**
+
+  ```js
+  Device.supportedABIs; // [ "arm64 v8", "Intel x86-64h Haswell", "arm64-v8a", "armeabi-v7a", "armeabi" ]
+  ```
+
+### Methods
+
+- `Device.hasNotch()`
+- `Device.hasSystemFeatureAsync(feature)` (Android only)
+- `Device.getIpAddressAsync()`
+- `Device.getMACAddressAsync()`
+- `Device.isAirplaneModeEnabledAsync()` (Android only)
+- `Device.isPinOrFingerprintSet()`
+- `Device.getFreeDiskStorageAsync()`
+- `Device.getUserAgentAsync()`
+- `Device.getTotalDiskCapacityAsync()`
+- `Device.getSerialNumberAsync()` (Android only)
+
+## Methods
+
+### `Device.hasNotch()`
+
+Tells if the device has a notch.
+
+#### Returns
+
+A boolean that represents the support for notch display.
+
+**Examples**
+
+```js
+import { hasNotch } from 'expo-device';
+
+hasNotch(); // true or false
+```
+
+### `Device.getUserAgentAsync()`
+
+Gets the device User Agent.
+
+#### Returns
+
+A Promise that resolves to a string of User Agent.
+
+**Examples**
+
+```js
+Device.getUserAgentAsync().then(userAgent => {
+  //Dalvik/2.1.0 (Linux; U; Android 9; Pixel 2 Build/PQ3A.190505.001)
+});
+```
+
+### `Device.getFreeDiskStorageAsync()`
+
+Gets available storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the storage size.
+
+**Examples**
+
+```js
+Device.getFreeDiskStorageAsync().then(storage => {
+  //5608296448
+});
+```
+
+### `Device.getTotalDiskCapacityAsync()`
+
+Gets full disk storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the full disk storage size.
+
+**Examples**
+
+```js
+Device.getTotalDiskCapacityAsync().then(storage => {
+  //17179869184
+});
+```
+
+### `Device.getIpAddressAsync()`
+
+Gets the device current IP address.
+
+#### Returns
+
+A Promise that resolves to a string of IP address.
+
+**Examples**
+
+```js
+Device.getIpAddressAsync().then(ip => {
+  // "92.168.32.44"
+});
+```
+
+### `Device.getMACAddressAsync(interfaceName?: string)`
+
+Gets the network adapter MAC address.
+
+#### Arguments (Android Only)
+
+- **interfaceName (_string_)** -- A string representing interface name (`eth0`, `wlan0`) or `null`, meaning the method should fetch MAC address of the first available interface. (On iOS this argument is ignored.)
+
+#### Returns
+
+A Promise that resolves to a string of the network adapter MAC address or empty string if there's no such address matching the interface.
```

Comment:
- This should return null instead of an empty string in the edge case scenarios.
- What if there is no interface with the given name?

Reply:
I'll return null if the mac address is null with the given interface name.
If there's no matching mac address with such interface name passed in, we'll reject the promise with message `Undefined interface name`.